### PR TITLE
Validates uniqueness

### LINKED ISSUE #3: validate_uniqueness_of
I saw that (at least in version 0.3.0) unique indexes are not translated into validations. Would that be a feature that would fit this gem?

### PR DESCRIPTION
**Pull request by kronn**
The "coming days" have come :-)
This fixes #3

### CODE REVIEW

File: spec/config/schema.rb [hunk 1]
```diff
@@ -32,5 +32,9 @@
     t.integer 'parent_id',                                           null: false
     t.integer 'kind',                                                null: false
     t.string 'list', array: true, limit: 3
+
+    t.index 'model', unique: true
+    t.index ['name', 'wheels'], unique: true
+    t.index ['doors'], unique: true, where: 'enabled = true'
```
Comment (codez) [inline spec/config/schema.rb:38]: This seems to break the sqlite tests. Can it be adjusted to work agnostically of the database?

**Review by codez — APPROVED** [review]
This looks really awesome! Thank you very much for the work. 

Could you also add an unique column to the example in the README, so the capability is promoted accordingly?

I will have a look on the 'out of range' test failure, so that Travis is happy before we merge.

**Comment by codez** [comment]
For some reason, the unique index on `wheels` is responsible for the "ActiveModel::RangeError: ... for ActiveModel::Type::Integer with limit 4" on Rails 5.0. If there is no obvious fix, it would be ok for me to disable the failing specs specifically for Rails 5.0. Can you have a look at that, too?

**Comment by kronn** [comment]
I am really happy for the extensive test-suite on this one. I'll try to tackle those "in the coming days" (SCNR).

**Comment by codez** [comment]
BTW: A changelog entry would be nice, too.

**Comment by codez** [comment]
Yo @kronn I herd you like uniqeueness so I put some uniqueuness in your uniqueness so we can go uniqueley unique. 

Are you up for a last review "in these days" before merging this into the main branch?

**Comment by kronn** [comment]
This looks good to me. Thanks for taking over, @codez

**Pull request merged**